src/definitions/modules/range.js

```diff
@@ -0,0 +1,273 @@
+                  pageX = mmEvent.originalEvent.touches[0].pageX;
+                } else {
+                  pageX = mmEvent.pageX;
+                }
+                value = module.determineValue(left, right, pageX);
+                if(pageX >= left && pageX <= right) {
+                  if(value >= settings.min && value <= settings.max) {
+                    module.setPosition(pageX - left - offset);
+                    module.setValue(value);
+                  }
+                }
+              }
+              var rangeMouseup = function(muEvent) {
+                if(isTouch) {
+                  $(document).off('touchmove', rangeMousemove);
+                  $(document).off('touchend', rangeMouseup);
+                } else {
+                  $(document).off('mousemove', rangeMousemove);
+                  $(document).off('mouseup', rangeMouseup);
+                }
+              }
+              if(isTouch) {
+                $(document).on('touchmove', rangeMousemove);
+                $(document).on('touchend', rangeMouseup);
+              }
+              else {
+                $(document).on('mousemove', rangeMousemove);
+                $(document).on('mouseup', rangeMouseup);
+              }
+            }
+          },
+          
+          setValuePosition: function(val, triggeredByUser) {
+            if(typeof triggeredByUser === 'undefined') {
+              triggeredByUser = true;
+            }
+            var position = module.determinePosition(val);
+            module.setPosition(position);
+            module.setValue(val, triggeredByUser);
+          },
+          
+          invoke: function(query) {
+            switch(query) {
+              case 'set value':
+                if(queryArguments.length > 0) {
+                  instance.setValuePosition(queryArguments[0], false);
+                }
+                break;
+            }
+          },
+          
+          addVisibilityListener: function(elem) {
+            
+            // Add a mutation observer (https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver)
+            // to detect when root invisible element is behing shown in order to initialize the thumb correctly
+            // when position and offets are available (https://stackoverflow.com/a/5974377)
+            
+            var observer = new MutationObserver(function(mutationList) {
+              if ($(elem).is(':visible')) {
+                observer.disconnect(); // Avoid infinite recursion because « module.setValuePosition » will trigger this observer
+                module.setValuePosition(settings.start);
+              }
+            });
+            
+            var closestHiddenParent = $(elem).parentsUntil(':visible');
+            
+            if (closestHiddenParent.length != 0) {
+              observer.observe(closestHiddenParent[closestHiddenParent.length - 1], {attributes: true});
+            }
+          },
+          
+        };
+        
```

Comment:
Not a blocking stuff, but we should add later common modules methods (debug, performance, display, verbose...).

Reply:
Good shout, didn't think of that